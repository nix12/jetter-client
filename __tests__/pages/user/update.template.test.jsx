import React from 'react';
import { shallow } from 'enzyme';

import Template from '../../../templates/user/update.template';

describe('<Update /> template', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      onUpdate: jest.fn(),
      error: null
    };

    wrapper = shallow(<Template {...props} />);
  });

  it('should render Update component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call inputChangedHandler to update state', () => {
    const mockedEvent = { target: { value: 'abc' } };

    wrapper.instance().inputChangedHandler(mockedEvent, 'password');
    expect(wrapper.state().controls.password.value).toEqual('abc');
  });

  it('should call submitHandler to register and direct new users', () => {
    const mockEvent = { preventDefault: jest.fn() };

    return wrapper
      .instance()
      .submitHandler(mockEvent)
      .then(() => {
        expect(props.onUpdate).toBeCalled();
      });
  });

  it('should render with no error', () => {
    expect(props.error).toEqual(null);
  });

  it('should render with error', () => {
    const mockEvent = { preventDefault: jest.fn() };

    props = {
      onUpdate: jest.fn(),
      error: 'Password update failed.'
    };

    wrapper = shallow(<Template {...props} />);

    return wrapper
      .instance()
      .submitHandler(mockEvent)
      .then(() => {
        expect(props.error).toEqual('Password update failed.');
        expect(wrapper.find('p').html()).toEqual(
          '<p>Password update failed.</p>'
        );
      });
  });
});
