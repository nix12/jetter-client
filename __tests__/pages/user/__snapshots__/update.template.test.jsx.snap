// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Update /> template should render Update component 1`] = `ShallowWrapper {}`;
